mss/L2Cache.cpp

```diff
@@ -0,0 +1,657 @@
+    ////////////////////////////////////////////////////////////////////////////////
+    // Regular Function/Subroutine Call
+    ////////////////////////////////////////////////////////////////////////////////
+
+    // Append L2Cache request queue for reqs from DCache
+    void L2Cache::appendDCacheReqQueue_(const olympia::InstPtr& inst_ptr) {
+        sparta_assert(dcache_req_queue_.size() <= dcache_req_queue_size_ ,"DCache request queue overflows!");
+
+        // Push new requests from back
+        dcache_req_queue_.emplace_back(inst_ptr);
+
+        ILOG("Append DCache->L2Cache request queue!");
+    }
+
+    // Append L2Cache request queue for reqs from ICache
+    void L2Cache::appendICacheReqQueue_(const olympia::InstPtr& inst_ptr) {
+        sparta_assert(icache_req_queue_.size() <= icache_req_queue_size_ ,"ICache request queue overflows!");
+
+        // Push new requests from back
+        icache_req_queue_.emplace_back(inst_ptr);
+
+        ILOG("Append ICache->L2Cache request queue!");
+    }
+
+    // Append BIU resp queue
+    void L2Cache::appendBIURespQueue_(const olympia::InstPtr& inst_ptr) {
+        sparta_assert(biu_resp_queue_.size() <= biu_resp_queue_size_ ,"BIU resp queue overflows!");
+
+        // Push new requests from back
+        biu_resp_queue_.emplace_back(inst_ptr);
+
+        ILOG("Append BIU->L2Cache resp queue!");
+    }
+
+    // Append DCache resp queue
+    void L2Cache::appendDCacheRespQueue_(const olympia::InstPtr& inst_ptr) {
+        sparta_assert(dcache_resp_queue_.size() <= dcache_resp_queue_size_ ,"DCache resp queue overflows!");
+        
+        // Push new resp to the dcache_resp_queue_
+        dcache_resp_queue_.emplace_back(inst_ptr);
+        ev_handle_l2cache_dcache_resp_.schedule(sparta::Clock::Cycle(0));
+        
+        ILOG("Append L2Cache->DCache resp queue!");
+    }
+
+    // Append ICache resp queue
+    void L2Cache::appendICacheRespQueue_(const olympia::InstPtr& inst_ptr) {
+        sparta_assert(icache_resp_queue_.size() <= icache_resp_queue_size_ ,"ICache resp queue overflows!");
+        
+        // Push new resp to the icache_resp_queue_
+        icache_resp_queue_.emplace_back(inst_ptr);
+        ev_handle_l2cache_icache_resp_.schedule(sparta::Clock::Cycle(0));
+        
+        ILOG("Append L2Cache->ICache resp queue!");
+    }
+
+    // Append BIU req queue
+    void L2Cache::appendBIUReqQueue_(const olympia::InstPtr& inst_ptr) {
+        sparta_assert(biu_req_queue_.size() <= biu_req_queue_size_ ,"BIU req queue overflows!");
+        
+        // Push new request to the biu_req_queue_ if biu credits are available with the L2Cache
+        if (biu_req_queue_.size() < biu_req_queue_size_) {
+            biu_req_queue_.emplace_back(inst_ptr);
+
+            ev_handle_l2cache_biu_req_.schedule(sparta::Clock::Cycle(0));
+        }
+        else {
+            sparta_assert("No space in biu_req_queue_! Why did the frontend issue push the request onto l2cache_pipeline_?");
```

Comment:
I didn't check all of the calls to `sparta_assert`, but your usage here is incorrect.  A string is _always_ true, so the assert will never fire.  Kinda surprised the compiler didn't pick up on that.  Suggested fixes:
```suggestion
            sparta_assert(false, "No space in biu_req_queue_! Why did the frontend issue push the request onto l2cache_pipeline_?");
            sparta_assert(!"No space in biu_req_queue_! Why did the frontend issue push the request onto l2cache_pipeline_?");

```

Reply:
Hmm! I thought it was always false by default. I picked it up from usage elsewhere in the Olympia model.
But good practice to explicitly mention it anyway! :)

Reply:
If you find it in other places in Olympia, please flag them -- that's a bug.  😉